utils/unpackage-app.js

```diff
@@ -0,0 +1,192 @@
+/*
+ This program and the accompanying materials are
+ made available under the terms of the Eclipse Public License v2.0 which accompanies
+ this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+ 
+ SPDX-License-Identifier: EPL-2.0
+ 
+ Copyright Contributors to the Zowe Project.
+*/
+
+const yauzl = require('yauzl');
+const fs = require('graceful-fs');
+const Promise = require('bluebird');
+const path = require('path');
+const packagingUtils = require('./packaging-utils');
+const serverUtils = require('../lib/util');
+const rmrf = require('rimraf');
+
+//assuming that this is file isnt being called from another that is already using the logger... else expect strange logs
+const logger = packagingUtils.coreLogger.makeComponentLogger("unpackage-app"); //should only need one for this program
+
+const argParser = require('../lib/argumentParser');
+const usage = 'Usage: --inputApp | -i INPUTAPP --outputDir | -o OUTPUTDIR --pluginsDir | -p PLUGINSDIR '
+      + '--zluxConfig | -c ZLUXCONFIGPATH [--verbose | -v] [--overwrite | -w]';
+
+//TODO if plugins get extracted read-only, then how would we go about doing upgrades? read-write for now!
+const FILE_WRITE_MODE = 0o600;
+const DIR_WRITE_MODE = 0o700;
+
+const OPTION_ARGS = [
+  new argParser.CLIArgument('inputApp', 'i', argParser.constants.ARG_TYPE_VALUE),
+  new argParser.CLIArgument('outputDir', 'o', argParser.constants.ARG_TYPE_VALUE),
+  new argParser.CLIArgument('pluginsDir', 'p', argParser.constants.ARG_TYPE_VALUE),
```

Comment:
It feels like one should drop the -p switch and just leave in the -c switch. Or, if the plan is to allow the person doing the install to control whether the plugin locator is added to site, instance, group or particular-user, then the switch should be about that choice and the zluxserver.json file would be used to choose the correct directory based on that choice.

Reply:
Just for flexibility in case the user knew the plugin dir but didn't know the location of the zlux config file, for some reason.